  * Tell liboobs to use md5 if pam the pam config specifies sha* as well
    as if it specifies md5. If we don't do this then the switch to sha512
    means that adding a users with system-tools-backends gives them a 3DES
    password. (LP: #287134)

 -- James Westby <james.westby@canonical.com>  Tue, 28 Oct 2008 21:06:26 +0000
--- a/Users/Users.pm
+++ b/Users/Users.pm
@@ -293,6 +293,7 @@
       foreach $i (@line)
       {
         $use_md5 = 1 if ($i eq "md5");
+        $use_md5 = 1 if ($i =~ /^sha\d+/);
       }
     }
   }
